File: implementations/php/src/postgres_serial.php
<?php

include "hashes.php";

try {
    $conn = new PDO("pgsql:host=127.0.0.1;port=5432;dbname=hashes", "root", "root123");
    $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    $conn->exec(<<<HEREA
            DROP TABLE IF EXISTS tb_php_serial;
            CREATE TABLE tb_php_serial
            (id SERIAL PRIMARY KEY,
            hash INTEGER NULL);
            HEREA);

    for ($i = 1; $i <= 10000; $i++) {
        $hash = hash32($i);
        $query = sprintf("INSERT INTO tb_php_serial(hash) VALUES (%d);", $hash);
        $conn->query($query);
    }
} catch (\Throwable $th) {
    if ($conn->inTransaction()) $conn->rollBack();
    echo $th->getMessage();
} finally {
    $conn = null;
}

File: implementations/java/src/PostgresSerial.java

import java.sql.*;

public class PostgresSerial {

    public static void main(String[] args) {
        int i, hash;
        try (
            Connection conn = DriverManager.getConnection(
                "jdbc:postgresql://127.0.0.1:5432/hashes",
                "root",
                "root123"
                );
            Statement stmt = conn.createStatement();
            PreparedStatement ps = conn.prepareStatement(
                "INSERT INTO tb_java_serial(hash) VALUES (?)"
            );
		) {
			stmt.executeUpdate("DROP TABLE IF EXISTS tb_java_serial;"
					+ "CREATE TABLE tb_java_serial "
					+ "(id SERIAL PRIMARY KEY, "
					+ "hash INTEGER NOT NULL)");

			for (i = 1; i <= 10000; i++) {
				hash = Hash.hash32(i);
				ps.setInt(1, hash);
                ps.executeUpdate();
			}
		} catch (Exception e) {
			System.err.println(e.getMessage());
		}
    }
}
